### DISCUSSION TITLE
Why there are multiple models are shown while I just input one?

### DISCUSSION THREAD

**Discussion by jimmyfu0729** [loco-3d/crocoddyl#1222] [Q&A]
I tried very simple urdf loading function but I don't know why there are many robots in the viewers are loaded. The urdf is from Solidworks urdf plug-in.

![截图 2024-02-10 17-58-56](https://github.com/loco-3d/crocoddyl/assets/56794191/f341f27f-97ff-40a3-9543-9a02b37c9e44)
![截图 2024-02-10 18-04-01](https://github.com/loco-3d/crocoddyl/assets/56794191/f5169a41-4828-4ee2-9414-0055f65ef8ba)
 The urdf file is shown(some) below:
 
<?xml version="1.0" encoding="utf-8"?>

<robot
  name="Robot_v4_corcoddyl">
  <link
    name="base_link">
    <inertial>
      <origin
        xyz="0.003449 -0.00015688 0.010763"
        rpy="0 0 0" />
      <mass
        value="0.22553" />
      <inertia
        ixx="0.00027207"
        ixy="5.4502E-10"
        ixz="-3.8793E-05"
        iyy="0.00015981"
        iyz="-2.666E-08"
        izz="0.00024608" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="/home/jimmy/crocoddyl/crocoddyl/Bipedal_Project/Robot_v4_corcoddyl/meshes/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="/home/jimmy/crocoddyl/crocoddyl/Bipedal_Project/Robot_v4_corcoddyl/meshes/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <link
    name="IMU_frame">
    <inertial>
      <origin
        xyz="1.2935E-07 -1.2935E-07 -0.001"
        rpy="0 0 0" />
      <mass
        value="0.0064185" />
      <inertia
        ixx="3.2008E-07"
        ixy="2.2628E-22"
        ixz="1.2633E-22"
        iyy="3.2008E-07"
        iyz="1.3741E-22"
        izz="6.3588E-07" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="/home/jimmy/crocoddyl/crocoddyl/Bipedal_Project/Robot_v4_corcoddyl/meshes/IMU_frame.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.89804 0.91765 0.92941 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="/home/jimmy/crocoddyl/crocoddyl/Bipedal_Project/Robot_v4_corcoddyl/meshes/IMU_frame.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="IMU"
    type="fixed">
    <origin
      xyz="0.047724 -0.00013922 0.01814"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="IMU_frame" />
    <axis
      xyz="0 0 0" />
  </joint>
  <link
    name="L1_frame">
    <inertial>
      <origin
        xyz="0.038799 0.0097245 -0.011065"
        rpy="0 0 0" />
      <mass
        value="0.16994" />
      <inertia
        ixx="0.00015096"
        ixy="-7.928E-08"
        ixz="6.4654E-06"
        iyy="9.9698E-05"
        iyz="-2.4526E-07"
        izz="0.00010829" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="/home/jimmy/crocoddyl/crocoddyl/Bipedal_Project/Robot_v4_corcoddyl/meshes/L1_frame.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.50196 0.50196 0.50196 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="/home/jimmy/crocoddyl/crocoddyl/Bipedal_Project/Robot_v4_corcoddyl/meshes/L1_frame.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="L1"
    type="revolute">
    <origin
      xyz="-0.05 0.045 -0.055"
      rpy="0 0 0" />
    <parent
      link="IMU_frame" />
    <child
      link="L1_frame" />
    <axis
      xyz="1 0 0" />
    <limit
      lower="-1.047"
      upper="1.047"
      effort="3.4355"
      velocity="5.236" />
    <safety_controller
      soft_upper_limit="1.045"
      soft_lower_limit="-1.045"
      k_velocity="0" />
  </joint>
  <link
    name="L2_frame">
    <inertial>
      <origin
        xyz="-8.6736E-19 1.5233E-17 -0.0067811"
        rpy="0 0 0" />
      <mass
        value="0.012997" />
      <inertia
        ixx="6.7497E-06"
        ixy="8.7283E-22"
        ixz="3.2595E-22"
        iyy="1.4566E-06"
        iyz="-9.0947E-22"
        izz="6.1406E-06" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="/home/jimmy/crocoddyl/crocoddyl/Bipedal_Project/Robot_v4_corcoddyl/meshes/L2_frame.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.89412 0.91373 0.92941 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="/home/jimmy/crocoddyl/crocoddyl/Bipedal_Project/Robot_v4_corcoddyl/meshes/L2_frame.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="L2"
    type="revolute">
    <origin
      xyz="0.04855 0.0193 -0.0618"
      rpy="0 0 0" />
    <parent
      link="L1_frame" />
    <child
      link="L2_frame" />
    <axis
      xyz="0 0 -1" />
    <limit
      lower="-1.047"
      upper="1.047"
      effort="3.4355"
      velocity="5.236" />
    <safety_controller
      soft_upper_limit="1.045"
      soft_lower_limit="-1.045"
      k_velocity="0" />
  </joint>
  <link
    name="L3_frame">
    <inertial>
      <origin
        xyz="4.4171E-05 0.024603 -0.032617"
        rpy="0 0 0" />
      <mass
        value="0.085431" />
      <inertia
        ixx="2.383E-05"
        ixy="6.4346E-21"
        ixz="1.4557E-10"
        iyy="1.2492E-05"
        iyz="9.5126E-23"
        izz="1.6935E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="/home/jimmy/crocoddyl/crocoddyl/Bipedal_Project/Robot_v4_corcoddyl/meshes/L3_frame.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.50196 0.50196 0.50196 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="/home/jimmy/crocoddyl/crocoddyl/Bipedal_Project/Robot_v4_corcoddyl/meshes/L3_frame.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="L3"
    type="revolute">
    <origin
      xyz="-4.5E-05 -0.0265 -0.0238"
      rpy="0 0 0" />
    <parent
      link="L2_frame" />
    <child
      link="L3_frame" />
    <axis
      xyz="0 0 0" />
    <limit
      lower="-1.047"
      upper="1.047"
      effort="4.4145"
      velocity="5.236" />
    <safety_controller
      soft_upper_limit="1.045"
      soft_lower_limit="-1.045"
      k_velocity="0" />
  </joint>
  <link

**Comment by jimmyfu0729** [comment]
I compared the urdf in the example_robot_data, it seems like there are just one mesh file path, I don't know if this problem is because my urdf contains mesh file path in every geometry description.
![图片](https://github.com/loco-3d/crocoddyl/assets/56794191/304e01c1-cc7a-4f42-aba6-9be96f19d6c3)
![图片](https://github.com/loco-3d/crocoddyl/assets/56794191/3e056310-1837-424a-a91a-dce0204dac15)

**Comment by cmastalli** [comment]
Hi @jimmyfu0729 

This issue has nothing to do with Crocoddyl. This is related with your own development using Meshcat. Please report issues or ask for helps in appropriate projects (i.e., Meshcat).
